scrapy/resolver.py

```diff
@@ -14,15 +16,23 @@ def __init__(self, reactor, cache_size, timeout):
         self.timeout = timeout
 
     def getHostByName(self, name, timeout=None):
-        if name in dnscache:
-            return defer.succeed(dnscache[name])
+        if PY2 and isinstance(name, text_type):
+            try:
+                safe_name = name.encode('idna')
+            except UnicodeError:
+                # we did our best, let it fail later.
+                safe_name = name
```

Comment:
So with same PY2 and text_type condition, trying idna encoding, but keep only `name`?